transcom/move.mil
# Add behat testing

Thanks for submitting a pull request! Below are a few things you can do to help us more quickly review your changes.

## Checklist

I have…

      against my changes


## Summary of Changes

This pull request…

- Adds behat testing for ReactJS tools presence verification
- builds the tools before the behat tests
- runs `npm test` in case we have testing, currently there are basic testing, but it is being executed already.

## Testing

To verify the changes proposed in this pull request…

1. remove your tool compiled files (build/static folder)
1. execute `circleci build --job behat`
1. You should see the tools being built
1. npm testing should be called 
1. test are passing

## Notes
I removed react-locator-map/src/App.test.js file because that testing was not passing and this PR wasn't about adding React Jest testing.
The reason why that test is not passing is there is a file that does not contain a class (or classes) but several functional components. I would suggest refactoring `web/modules/custom/react_tools/tools/apps/react-locator-map/src/components/listItem.js` to a class or separate classes so that App.js can be testable.
